docs: Added some docs and githubpage generation (#95)
numaproj/numaflow: docs/QUICK_START.md

# Quick Start

To see how NumaFlow works, you can install it, and run a simple pipeline, which contains a source vertex to generate messages, and a function vertex that echos the messages, and a log sink printing the messages.

## Prerequisites

A Kubernetes cluster is needed to try out NumaFlow. If needed, you can create a cluster locally using
[`k3d`](https://k3d.io/) as below. You also need to get [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/) installed.

```shell
# Create a cluster with default name k3s-default
k3d cluster create -i rancher/k3s:v1.21.7-k3s1

# Get kubeconfig for the cluster
k3d kubeconfig get k3s-default
```

## NumaFlow Installation

```sh
kubectl create ns numaflow-system
kubectl apply -n numaflow-system -f https://raw.githubusercontent.com/numaproj/numaflow/stable/config/install.yaml

```

Open a port-forward so you can access the UI on [https://localhost8443](https://localhost8443).

```sh
kubectl -n numaflow-system port-forward deployment/numaflow-server 8443:8443
```

## A Simple Pipeline

> [Inter-Step Buffers](./INTER_STEP_BUFFER.md) are esstienal to run NumaFlow pipelines, to get Inter-Step Buffers in place, an [Inter-Step Buffer Service](./INTER_STEP_BUFFER_SERVICE.md) is required. We use [Nats JetStream](https://docs.nats.io/nats-concepts/jetstream) as a `Inter-Step Buffer Service` provider in this example.

Reviewer: ```suggestion
[Inter-Step Buffers](./INTER_STEP_BUFFER.md) are essential to run NumaFlow pipelines, to get Inter-Step Buffers in place, an [Inter-Step Buffer Service](./INTER_STEP_BUFFER_SERVICE.md) is required. We use [Nats JetStream](https://docs.nats.io/nats-concepts/jetstream) as a `Inter-Step Buffer Service` provider in this example.
```

Author: done.

```shell
kubectl apply -f ./examples/0-isbsvc-jetstream.yaml
```

> After all the isbsvc pods are up, create a simple pipeline, and you can see the pipeline topoligy in the UI.

Reviewer: ```suggestion
After all the isbsvc pods are up, create a simple pipeline, and you can see the pipeline topology in the UI.
```

Author: done.

2021/12/17 22:14:35 (simple-pipeline-output) {"Data":"6FmrNo6qwRY=","Createdts":1639779267118848488}
```

> You can delete the pipeline by using

Reviewer: ```suggestion
You can delete the pipeline by
```

for consistency with the next line

Author: done.

```shell
kubectl delete -f ./examples/1-simple-pipeline.yaml
```

The `Inter-Step Buffer Service` can be deleted by

```shell
kubectl delete -f ./examples/0-isbsvc-jetstream.yaml
```

## What's Next

> After exploring how a NumaFlow pipeline runs, you can check what data [Sources](./sources/GENERATOR.md) and [Sinks](./sinks/KAFKA.md) that NumaFlow supports, or learn how to write [User Defined Functions](./USER_DEFINED_FUNCTIONS.md).

Reviewer: ```suggestion
After exploring how a NumaFlow pipeline runs, you can check what data [Sources](./sources/GENERATOR.md) and [Sinks](./sinks/KAFKA.md) NumaFlow supports out of the box, or learn how to write [User Defined Functions](./USER_DEFINED_FUNCTIONS.md).
```

Author: done.
